File: semester_3_(fall-22)/Object Oriented Programming Lab/Lab#1/Task#3.cpp
/*
Create a class called BankAccount that models a checking account at a bank. The program creates an
account with an opening balance, displays the balance, makes a deposit and a withdrawal, and then
displays the new balance. Note in withdrawal function, if balance is below Rs. 500 then display message
showing insufficient balance otherwise allow withdrawal.
*/
#include <iostream>
using namespace std;

class BankAccount{
    private:
        int balance;
    public:
        //Constructing initially to 1000:
        BankAccount():balance(1000) {}
        void deposit(int d){
            balance += d;
        }
        void withdrawl(int w){
            if (balance <= 500)
                cout << "\n\t**Insufficient Balance**" ;
            else
                balance -= w;
        }
        void show(){
            cout << "\nCurrent Balance:\t" << balance ;
        }
};

int main(){
    BankAccount b1;
    b1.show();
    b1.deposit(500);
    b1.show();
    b1.withdrawl(500);
    b1.show();
    b1.withdrawl(500);
    b1.show();
    b1.withdrawl(500);
    b1.show();

    return 0;
}

File: semester_3_(fall-22)/Object Oriented Programming Lab/Lab#1/Task#3.py
'''
Create a class called BankAccount that models a checking account at a bank. The program creates an
account with an opening balance, displays the balance, makes a deposit and a withdrawal, and then
displays the new balance. Note in withdrawal function, if balance is below Rs. 500 then display message
showing insufficient balance otherwise allow withdrawal.
'''
class BankAccount:
    def __init__(self, a = 1000):
        self.balance = a

    def deposit(self, d):
        self.balance += d

    def withdrawal(self, w):
        if (self.balance <= 500):
            print("\t**Insufficient Balance**") ;
        else:
            self.balance -= w

    def show(self):
        print("Current Balance:\t", self.balance)

b1 = BankAccount()
b1.show()
b1.deposit(500)
b1.show()
b1.withdrawal(500)
b1.show()
b1.withdrawal(500)
b1.show()
b1.withdrawal(500)
b1.show()